src/api/BitbucketVersionControl.ts

```diff
@@ -61,52 +61,234 @@ export default class BitbucketVersionControl implements VersionControlAbstractio
+					'BitbucketVersionControl: could not create working of branch');
+		}
+	}
+
+	/**
+	 * Internal function to get the last commit in a branch of the repository
+	 * @param branchName The name of the branch where you want to get the last commit from
+	 * @returns The ID of the last commit in the branch
+	 */
+	private async getLastCommitId(branchName: string): Promise<string> {
 		const bitbucketApiAuthHeader = {
 			auth: {
 				username: this.authentication.user,
 				password: this.authentication.password
 			}
 		};
 		this.loggerReference.logMessage(LogLevels.debug, 'BitbucketVersionControl: fetching last commit information');
-		let lastCommitDetails = await axios.get(
-			`${this.serverConfig.serverUrl}/rest/api/1.0/projects/${this.serverConfig.projectName}/repos/${this.serverConfig.repositoryName}/commits?until=${this.serverConfig.branchName}&limit=0&start=0`,
-			bitbucketApiAuthHeader
-		);
-		// ToDo: 404 in case of non existent branch -> {"errors": [{"context": null,"message": "Commit 'feature/directus-changes' does not exist in repository 'directus-plugin-playground'.","exceptionName": "com.atlassian.bitbucket.commit.NoSuchCommitException"}]}
-		// Las commit is always filled as long as there is at least one in the repo; even on new branches you get the last commit ID from the origin
-		if (lastCommitDetails.status !== 200) {
-			this.loggerReference.logMessage(LogLevels.error, 'could not fetch the ID of the latest commit from Bitbucket');
-			// ToDo: error handling in combination with detection if file is there
-			return false;
+		let response: any = undefined;
+		try {
+			const requestResponse = await axios.get(
+				`${this.serverConfig.serverUrl}/rest/api/1.0/projects/${this.serverConfig.projectName}/repos/${this.serverConfig.repositoryName}/commits?until=${branchName}&limit=0&start=0`,
+				bitbucketApiAuthHeader
+			);
+			response = requestResponse; 
+		} catch (error: any) {
+			this.loggerReference.logObject(LogLevels.debug, 'Axios error result:', error.response.data.errors);
 		}
-		const lastCommitId = lastCommitDetails.data.values[0].id;
-		this.loggerReference.logMessage(LogLevels.debug, `BitbucketVersionControl: latest commit in Directus Changelog Branch: ${lastCommitId}`);
+		// Last commit is always filled as long as there is at least one in the repo; even on new branches you get the last commit ID from the origin
+		if (response.status !== 200) {
+			throw this.exceptionProvider.getNewException(ExceptionTypes.UnexpectedResponseException,
+					'BitbucketVersionControl: could not fetch the ID of the latest commit in the branch');
+		}
+		return response.data.values[0].id as string;
+	}
 
+	/**
+	 * Internal function to get the content of the current changelog in the repository
+	 * @returns An array os strings, each item represents a line in the file
+	 */
+	private async getCurrentChangelogContent(): Promise<Array<string>> {
+		let result = new Array<string>();
+		const bitbucketApiAuthHeader = {
```

Comment:
looks likethis auth header could be reused

Reply:
makes sense, implemented